docs/guide/troubleshooting.md

```diff
@@ -65,3 +65,18 @@ If you are running Vite with WSL2, Vite cannot watch file changes in some condit
 If HMR is not handled by Vite or a plugin, a full reload will happen.
 
 Also if there is a dependency loop, a full reload will happen. To solve this, try removing the loop.
+
+## Others
+
+### Syntax Error / Type Error happens
+
+Vite cannot handle and does not support code that only runs on non-strict mode (sloppy mode).
+This is because Vite uses ESM and it is always [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode) inside ESM.
```

Comment:
```suggestion
Vite cannot handle and does not support code that only runs on non-strict mode (sloppy mode). This is because Vite uses ESM and it is always [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode) inside ESM.
```